drivers/include/max31865.h

```diff
@@ -0,0 +1,276 @@
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#pragma once
+
+/**
+ * @defgroup    drivers_max31865 MAX31865 RTD-to-Digital converter driver
+ * @ingroup     drivers_sensors
+ * @ingroup     drivers_saul
+ * @brief       Driver for the SPI RTD-to-Digital converter MAX31865.
+ *
+ * \section sec_max31865_ovrvw Overview
+
+ * The MAX31865 is a resistance-to-digital
+ * converter optimized for platinum resistance temperature
+ * detectors (RTDs). An external resistor sets the sensitivity
+ * for the RTD being used and a precision delta-sigma ADC
+ * converts the ratio of the RTD resistance to the reference
+ * resistance on the board, to a 15-bit word.
+ *
+ * - Compatible with 2-, 3-, and 4-Wire Sensor Connections
+ * - 15-Bit ADC resolution; resolution 0.03125°C
+ * - Total accuracy over all operating conditions: 0.5°C (0.05% FS) max
+ * - 21 ms max conversion time
+ *
+ * The RTD register and the threshold registers represent the ratio
+ * of the RTD resistance to the reference resistance.
+ *
+ * @note See the
+ * <a href="https://www.analog.com/media/en/technical-documentation/data-sheets/MAX31865.pdf">
+ * datasheet</a> for more information.
+ *
+ * \section sec_max31865_lut Lookup table
+ *
+ * In order to convert the ADC code to temperature with a decent precision
+ * and low CPU usage by using fixed point computation, this driver uses a lookup table.
+ * The values of the lookup table depend on the type of RTD, the reference
+ * resistance on the board and the temperature range.
+ * A default lookup table for a Pt100 sensor, a reference resistance of 330&Omega;
+ * and a temperature range of -200..+650°C is provided.
+ * With this lookup table, the standard deviation of the computation
+ * error over the -200..+650°C range is 0.011°C and the maximum error
+ * is 0.039°C.
+ *
+ * If the default lookup doesn't fit the application, the user can generate
+ * a custom one with the Python script `drivers/max31865/dist/genlut.py`.
+ * Type `genlut.py -h` for help.
+ * Move the generated header file to the application project directory and
+ * pass the custom lookup table to the device parameters.
+ *
+ * @code
+ * #include "max31865_mylut.h"      // header generated by genlut.py
+ *
+ * max31865_params_t my_params = max31865_params[0];
+ * my_params.lut = &mylut;
+ * max31865_init(&dev, &my_params);
+ * @endcode
+ *
+ * \section sec_max31865_usage Usage
+ *
+ * The default configuration is set in \ref max31865_params.h as a
+ * \ref max31865_params_t structure.
+ * To change it, pass a different value to the \a params argument of
+ * #max31865_init().
+ * In particular, let the SPI device and chip select pin fit the board.
+ *
+ * @code
+ * max31865_params_t my_params = max31865_params[0];
+ * my_params.spi = SPI_DEV(1);
+ * my_params.cs_pin = GPIO_PIN(0, 5);
+ * max31865_init(&dev, &my_params);
+ * @endcode
```

Comment:
This is not the preferred way to change parameters, especially the interface parameters. Defines exist for this purpose, otherwise these defines wouldn't make sense. The user has different options for changing them using these defines. They can be set either
1. in `board.h` for a board that has a MAX32865 on board or
2. in the Makefile of an application's using the `CFLAGS` variable or
3. in the make command also using the `CFLAGS` variable.

```suggestion
 * Most of them can be changed using the preprocessor definitions either
 * in the board definition in `board.h` or in the Makefile of the application
 * using the `CFLAGS` variable.
 * In particular, let the SPI device and chip select pin fit the board.
 *
 * @code
 * define MAX31865_PARAM_SPI        (SPI_DEV(0))
 * define MAX31865_PARAM_CS_PIN     (GPIO_PIN(0, 5))
 * @endcode
```